glusterd2/commands/volumes/volume-profile.go

```diff
@@ -0,0 +1,312 @@
+package volumecommands
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/servers/sunrpc/dict"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+
+	"github.com/gorilla/mux"
+)
+
+func registerVolProfileStepFuncs() {
+	transaction.RegisterStepFunc(txnVolumeProfile, "volume.Profile")
+}
+
+func volumeProfileHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	volname := mux.Vars(r)["volname"]
+	option := mux.Vars(r)["option"]
+
+	txn, err := transaction.NewTxnWithLocks(ctx, volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+	defer txn.Done()
+
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "volume must be in start state")
+		return
+	}
+
+	if !getActiveProfileSession(volinfo) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "there are no active profile sessions running")
+		return
+
+	}
+
+	txn.Steps = []*transaction.Step{
+		{
+			DoFunc: "volume.Profile",
+			Nodes:  volinfo.Nodes(),
+		},
+	}
+
+	if err := txn.Ctx.Set("option", option); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	if err := txn.Ctx.Set("volinfo", volinfo); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+	if err := txn.Do(); err != nil {
+		logger.WithError(err).WithField(
+			"volume", volname).Error("transaction to profile volume failed")
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+
+	var volumeProfileInfo []BrickProfileInfo
+	for _, node := range volinfo.Nodes() {
+		var nodeResult []map[string]string
+		err := txn.Ctx.GetNodeResult(node, "node-result", &nodeResult)
+		if err != nil {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+			return
+		}
+		// Get profile Info Array for  each nodes of a volume
+		for brickResult := range nodeResult {
+			var brickProfileInfo BrickProfileInfo
```

Comment:
can you rename the variable to have a shorter name something like ```bp```
because the assignment line looks slightly lengthy.

Reply:
@Madhu-1  Yes the assignment line looks lengthy but I am more inclined towards having a good variable name in order to make the code easier to understand.